File: jlc_clojure/test/glue_paths/jukebox/step_definitions/is_it_friday_yet.clj
(ns glue-paths.jukebox.step-definitions.is-it-friday-yet
  (:require [fundingcircle.jukebox :refer [step]]))

(defn is-it-friday?
  [day]
  (= day "Friday"))

(step "today is Sunday"
  [board]
  (assoc board :today "Sunday"))

(step "I ask whether it's Friday yet"
  [board]
  (assoc board :actual-answer (is-it-friday? (:today board))))

(step " I should be told {string}"
  [board expected-answer]
  (assert (= (:actual-answer board) expected-answer)))

File: jlc_ruby/spec/glue_paths/jukebox/step_definitions/is_it_friday_yet.rb
# frozen_string_literal: true

require 'jukebox'

module IsItFriday
  extend Jukebox

  def is_it_friday(day)
    day == 'Friday'
  end

  step 'today is Sunday' do |board|
    board.merge(today: 'Sunday')
  end

  step 'I ask whether it\'s Friday yet' do |board|
    board.merge(actual_answer: is_it_friday(board[:today]))
  end

  step 'I should be told {string}' do |board, expected_answer|
    expect(board[:actual_answer]).to eq(expected_answer)
  end
end
